src/registry.rs

```diff
@@ -218,18 +220,21 @@ impl Registry {
 /// ```
 #[derive(Debug)]
 pub struct SystemRegistry {
-    system: Addr<Arbiter>,
-    registry: InnerRegistry,
+    system: Arbiter,
+    registry: HashMap<TypeId, Box<Any + Send>>,
 }
 
-type AnyMapSend = HashMap<TypeId, Box<Any + Send>, BuildHasherDefault<FnvHasher>>;
-type InnerRegistry = Arc<ReentrantMutex<RefCell<AnyMapSend>>>;
+lazy_static::lazy_static! {
+    static ref SREG: ReentrantMutex<RefCell<HashMap<usize, SystemRegistry>>> = {
```

Comment:
Since it is guarded, do we actually need `RefCell`? We could get away without overhead of its runtime borrow check

Reply:
we can not this is reentrant mutex, so several owners

Reply:
isn't several owners are possible only within the same thread?

Reply:
several read-only owners

Reply:
ok